QuantumExperience/QuasmDriver.cs

```diff
@@ -0,0 +1,171 @@
+        }
+
+        /// <summary>
+        /// Ingnored becuase this machine doesn't have it. And we recycle qubits anyway
+        /// </summary>
+        public class QSimReset : Reset
+        {
+            public QSimReset(IOperationFactory m) : base(m)
+            {
+            }
+
+            public override Func<Qubit, QVoid> Body
+            {
+                get
+                {
+                    return delegate (Qubit q)
+                    {
+                        return QVoid.Instance;
+                    };
+                }
+            }
+        }
+
+        public class QSimCCNOT : CCNOT
+        {
+            public QSimCCNOT(IOperationFactory m) : base(m)
+            {
+            }
+
+            public override Func<(Qubit, Qubit, Qubit), QVoid> Body
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+        }
+
+        public class QSimCNOT : CNOT
+        {
+            public QSimCNOT(IOperationFactory m) : base(m)
+            {
+            }
+
+            public override Func<(Qubit, Qubit), QVoid> Body
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+        }
+
+        public class QSimM : M
+        {
+            public QSimM(IOperationFactory m) : base(m)
+            {
+            }
+
+            public override Func<Qubit, Result> Body
+            {
+                get
+                {
+                    return delegate (Qubit q)
+                    {
+                        if (q == null)
+                        {
+                            return Result.Zero;
+                        }
+                        QuasmLog.AppendLine("measure q[" + (uint)q.Id + "] -> c[" + (uint)q.Id + "];");
+                        
+                        Console.WriteLine("");
+                        Console.WriteLine("QUASM file");
+                        Console.Write(QuasmLog.ToString());
+                        Console.WriteLine("");
+                        var result = (Factory as QuasmDriver).RunQuasm(QuasmLog);
+                        return result[q.Id];
+                    };
+                }
+            }
+        }
+
+        public class QSimMeasure : Measure
+        {
+            public QSimMeasure(IOperationFactory m) : base(m)
+            {
+            }
+
+            public override Func<(QArray<Pauli>, QArray<Qubit>), Result> Body => throw new NotImplementedException();
+        }
+
+        public class QSimX : X
+        {
+            public QSimX(IOperationFactory m) : base(m)
+            {
+            }
+
+            public override Func<Qubit, QVoid> Body
+            {
+                get
+                {
+                    return delegate (Qubit q1)
+                    {
+                        if (q1 == null)
```

Comment:
doesn't this need to be logged in the quasm.log?

Reply:
Not yet used in the example. But indeed; Should have been for YXZ.